Sources/Fluent/Query/QueryRepresentable.swift

```diff
@@ -107,6 +107,7 @@ extension QueryRepresentable where Self: ExecutorRepresentable {
 
         if let _ = entity.id, entity.exists {
             // update
+            try type(of: entity).willUpdate(entity: entity)
```

Comment:
Actually, I think you might be able to use `E.willUpdate(...)` here. Can you try that @Evertt?